renku_notebooks/api/classes/server.py

```diff
@@ -161,219 +167,439 @@ def _get_image(self, image):
         self.using_default_image = verified_image == current_app.config["DEFAULT_IMAGE"]
         return verified_image, is_image_private
 
-    def _create_registry_secret(self):
+    def _get_registry_secret(self, b64encode=True):
         """If an image from gitlab is used and the image is not public
         create an image pull secret in k8s so that the private image can be used."""
-        secret_name = f"{self.safe_username}-registry-{str(uuid4())}"
-        git_host = urlparse(current_app.config.get("GITLAB_URL")).netloc
-        gitlab_project_id = self._user.gitlab_client.projects.get(
-            f"{self.namespace}/{self.project}"
-        ).id
         payload = {
             "auths": {
                 current_app.config.get("IMAGE_REGISTRY"): {
                     "Username": "oauth2",
-                    "Password": self._user.oauth_token,
-                    "Email": str(
-                        self._user.hub_user.get("auth_state", {})
-                        .get("gitlab_user", {})
-                        .get("email")
-                    ),
+                    "Password": self._user.git_token,
+                    "Email": self._user.email,
                 }
             }
         }
-
-        data = {
-            ".dockerconfigjson": base64.b64encode(json.dumps(payload).encode()).decode()
+        output = json.dumps(payload)
+        if b64encode:
+            return base64.b64encode(output.encode()).decode()
+        return output
+
+    def _get_session_k8s_resources(self):
+        cpu = float(self.server_options["cpu_request"])
+        mem = self.server_options["mem_request"]
+        gpu_req = self.server_options.get("gpu_request", {})
+        gpu = {"nvidia.com/gpu": str(gpu_req)} if gpu_req else None
+        resources = {
+            "requests": {"memory": mem, "cpu": cpu},
+            "limits": {"memory": mem, "cpu": cpu},
```

Comment:
limiting the cpu is a departure from our current practice (we make it burstable usually by default). I think we should keep the current behavior for the time being but in a follow-up we could offer a setting (via the values file) to enforce cpu limits as well.

Reply:
I think this is why starting the sessions is slow. My bad. I just completely overlooked the fact that the cpu should not be limited and only be part of the requests. I just copied and pasted the code for both requests and limits. I will change this - it was not a deliberate change.

Reply:
done only memory is limited now